### PR TITLE
revise .' deprecation message and add related TODOs to base/deprecated.jl

### PR DESCRIPTION
**Pull request by Sacha0**
This pull request revises the `.'` deprecation message in accord with changes in #25364 and #25461, and adds `.'`-deprecation related TODOs to base/deprecated.jl's 0.7 section. Best!

### CODE REVIEW

File: src/julia-syntax.scm [hunk 1]
```diff
@@ -2349,28 +2349,31 @@
...
+                                             "version of `A` (often by wrapping in the `Transpose` type), "
+                                             "and `copy(transpose(A))` which lazily transposes "
+                                             "`A` as above and then materializes that lazily "
+                                             "transposed `A` into a freshly allocated matrix "
+                                             "of `A`'s type. Which rewrite is appropriate depends on "
                                              "context: If `A.'` appears in a multiplication, "
                                              "left-division, or right-division operation that "
                                              "was formerly specially lowered to an `A_mul_B`-like "
-                                             "call, then the lazy `Tranpose(A)` is the correct "
+                                             "call, then the lazy `tranpose(A)` is the correct "
```
Comment (fredrikekre) [inline src/julia-syntax.scm]: `tranpose` -> `transpose`

**Comment by mbauman** [comment]
Does this message really need to be so dang long?  I think users would be just fine with:

> Postfix `x.'` for array transpose is deprecated. Use `transpose(x)` or `copy(transpose(x))` instead.

**Comment by Sacha0** [comment]
> Does this message really need to be so dang long? I think users would be just fine with:
>>Postfix x.' for array transpose is deprecated. Use transpose(x) or copy(transpose(x)) instead.

On the one hand I sympathize with the desire to make this message short. On the other hand I feel the complete information is useful. (Absent discussion of the eager->lazy transition and lowering change, whether `transpose(A)` or `copy(transpose(A))` is the appropriate rewrite is not clear. Encountering a terse deprecation message (lacking guidance about the choice) could be frustrating as a user.)

To both keep this message short and provide the aforementioned guidance, one could direct the user to a complete news entry from the message. That approach seems reasonable, though I worry about adding extra hoops (needing to locate NEWS.md and search for the appropriate entry) that a substantial fraction of users may either jump through but be frustrated by or not jump through and then be frustrated when rewrites don't work as desired. The expected volume of frustration with the additional hoops struck me as greater than the expected volume of frustration with a long deprecation message, so I went with the latter. (I wrote this deprecation message rapidly, and imagine we can tighten it up a bit without losing much if any content besides.)

Thoughts? Thanks! :)

File: src/julia-syntax.scm (hunk 1 above)
Comment (Sacha0) [reply src/julia-syntax.scm]: Fixed. Thanks! :) (Also added a `[ci skip]`.)

**Comment by Sacha0** [comment]
(Independent of how the deprecation message should be more extensively revised, I favor not coupling these fixes to those more extensive revisions, if agreeable :).)

**Comment by mbauman** [comment]
That's fair. I'll propose a simplification separately.

**Review by mbauman — APPROVED** [review]

**Comment by Sacha0** [comment]
Thanks all! :)

**Pull request merged**